File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    let answer = 1;
    
    for (let i = 1; i <= n / 2 + 1; i++) {
        let sum = 0;
        sum += i;
        for (let j = i + 1; j < n / 2 + 1; j++) {
            sum += j;
            if (sum > n) break;
            if (sum === n) answer++;
        }
    }
         
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    
    for start in range(1, n // 2 + 1):
        hap = 0
        for i in range(start, n // 2 + 2):
            hap += i
            
            if hap > n:
                break
            if hap == n:
                answer += 1
        
    return answer + 1